BrightFutures/Future.swift

```diff
@@ -383,12 +379,7 @@ public extension Future {
     /// Returns self
     public func onSuccess(context c: ExecutionContext = executionContextForCurrentContext(), callback: SuccessCallback) -> Future<T> {
         self.onComplete(context: c) { result in
-            switch result {
-            case .Success(let val):
-                callback(val.value)
-            default:
-                break
-            }
+            result.analysis(ifSuccess: callback, ifFailure: { _ in })
```

Comment:
I've replaced a lot of these switches with `Result.analysis`. It looks weird at first, but is really convenient when understood completely. These changes are independent of the replacement of `Result` (they can be discarded, if needed)